src/formatDate.js

```diff
@@ -9,6 +9,39 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const dataArr = date.split(fromSeparator);
+  const obj = {};
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    obj[fromFormat[i]] = dataArr[i];
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (const key in obj) {
+      if (toFormat[i] === key) {
+        result.push(obj[key]);
+      }
+    }
+
+    if (toFormat[i] === 'YY') {
+      result.push(obj['YYYY'].split('').slice(2).join(''));
+    }
+
+    if (toFormat[i] === 'YYYY') {
+      if (+obj['YY'] < 30) {
+        result.push(`20${obj['YY']}`);
+      }
+
+      if (+obj['YY'] >= 30) {
+        result.push(`19${obj['YY']}`);
+      }
+    }
+  }
+
+  return result.join(toSeparator);
```

Comment:
The function logic is mostly correct, but there are a few issues to address. Firstly, the loop in lines 22-41 is inefficient because it contains a nested loop that iterates over the `obj` keys for every item in `toFormat`. This can be simplified by directly accessing the corresponding value from the `obj` using the key. Secondly, the conditions for converting `YY` to `YYYY` and vice versa are not correctly implemented. When converting `YY` to `YYYY`, you should check if the `YY` part exists in the `obj` before attempting to convert. The conversion logic should only be applied if the `YY` or `YYYY` keys are present in the `toFormat` array. Lastly, the function does not handle the case where `YYYY` is converted to `YY` if the `fromFormat` contains `YYYY` and the `toFormat` contains `YY`. This should be handled outside the loop to avoid redundant checks.